File: api/collegeportal/src/main/java/com/collegeportal/modules/auth/service/AuthService.java
package com.collegeportal.modules.auth.service;

import com.collegeportal.modules.auth.dto.request.CompleteProfileRequestDTO;
import com.collegeportal.modules.auth.dto.request.LoginRequestDTO;
import com.collegeportal.modules.auth.dto.request.RegisterRequestDTO;
import com.collegeportal.modules.auth.dto.request.ForgotPasswordRequestDTO;
import com.collegeportal.modules.auth.dto.request.ResetPasswordRequestDTO;
import com.collegeportal.modules.auth.dto.request.GoogleAuthRequestDTO;
import com.collegeportal.modules.auth.dto.response.AuthResponseDTO;

public interface AuthService {

    AuthResponseDTO register(RegisterRequestDTO request);

    AuthResponseDTO login(LoginRequestDTO request);
    
    AuthResponseDTO forgotPassword(ForgotPasswordRequestDTO request);
    
    AuthResponseDTO resetPassword(ResetPasswordRequestDTO request);

    AuthResponseDTO googleAuth(GoogleAuthRequestDTO request);

    AuthResponseDTO googleRegister(String accessToken, RegisterRequestDTO request);

    AuthResponseDTO completeProfile(String email, CompleteProfileRequestDTO request);
}

File: web/src/services/authService.js
import httpClient from './httpClient';

const login = (data) => httpClient.post('/api/v1/auth/login', data).then((res) => res.data);
const register = (data) => httpClient.post('/api/v1/auth/register', data).then((res) => res.data);
const googleAuth = (idToken) => httpClient.post('/api/v1/auth/google', { idToken }).then((res) => res.data);
const googleRegister = (accessToken, data) => httpClient.post('/api/v1/auth/google/register', { accessToken, ...data }).then((res) => res.data);

export default { login, register, googleAuth, googleRegister };
